src/test/unit/marker-data.test.js

```diff
@@ -661,3 +678,91 @@ describe('filterRawMarkerTableToRange', () => {
     ]);
   });
 });
+
+// We don't need to test with other marker types since they are already being
+// tested in `filterRawMarkerTableToRange` tests.
+describe('filterRawMarkerTableToRangeWithMarkersToDelete', () => {
+  function setup(
+    markers: Array<[string, Milliseconds, null | Object]>
+  ): Thread {
+    markers = markers.map(([name, time, payload]) => {
+      if (payload) {
+        // Force a type 'DummyForTests' if it's inexistant
+        payload = { type: 'DummyForTests', ...payload };
+      }
+      return [name, time, payload];
+    });
+
+    // Our marker payload union type is too difficult to work with in a
+    // generic way here.
+    return getThreadWithMarkers((markers: any));
+  }
+
+  it('filters generic markers without markerToDelete', () => {
+    const markers = [
+      ['0', 0, null],
+      ['1', 1, null],
+      ['2', 2, null],
+      ['3', 3, null],
+      ['4', 4, null],
+      ['5', 5, null],
+      ['6', 6, null],
+      ['7', 7, null],
+    ];
+    const { markers: markerTable } = setup(markers);
+    const filteredMarkerTable = filterRawMarkerTableToRangeWithMarkersToDelete(
+      markerTable,
+      [],
+      { start: 2.3, end: 5.6 }
+    );
+    // Note: because the test fixture utility adds the strings in order, the
+    // string indices are actually the same as the name themselves, which make
+    // it possible to do an easy and readable assertion.
+    expect(filteredMarkerTable.name).toEqual([3, 4, 5]);
+  });
+
+  it('filters generic markers with markerToDelete', () => {
+    const markers = [
+      ['0', 0, null],
```

Comment:
'A', 'B', 'C', etc wouldn't help in this case because marker.name array only store string indexes to stringTable. So we still get the indexes here instead of string itself. See the comment here: 
https://github.com/firefox-devtools/profiler/pull/1790/files#diff-8d13052b5f9535584a9b3c9fd1f2d715R718